File: java-vfs/src/main/java/emg/java/vfs/extensions/ListExtensions.java
package emg.java.vfs.extensions;

import java.util.List;

public class ListExtensions {
    public static <T> List<T> tail(List<T> list) {
        return list.subList(1, list.size());
    }

    public static <T> T head(List<T> list) {
        return list.get(0);
    }
}

File: kotlin-vfs/src/main/kotlin/emg/kotlin/vfs/extensions/ListExtensions.kt
package emg.kotlin.vfs.extensions


fun <T> List<T>.head(): T {
    return this[0]
}

fun <T> List<T>.tail(): List<T> {
    return this.subList(1, this.size)
}

fun <T> MutableList<T>.init(): MutableList<T> {
    return this.subList(0, this.size - 1)
}
